packages/common/src/services/audius-backend/solana.ts

```diff
@@ -757,3 +713,210 @@ export const createVersionedTransaction = async (
+ *
+ * Note that this uses payment router to do the transfer, so that indexing sees
+ * this transfer and handles it appropriately.
+ */
+export const recoverUsdcFromRootWallet = async ({
+  sdk,
+  sender,
+  receiverEthWallet,
+  amount
+}: {
+  sdk: AudiusSdk
+  sender: Keypair
+  receiverEthWallet: string
+  amount: bigint
+}) => {
+  const { userBank } =
+    await sdk.services.claimableTokensClient.getOrCreateUserBank({
+      ethWallet: receiverEthWallet,
+      mint: 'USDC'
+    })
+
+  // See: https://github.com/solana-labs/solana-program-library/blob/d6297495ea4dcc1bd48f3efdd6e3bbdaef25a495/memo/js/src/index.ts#L27
+  const memoInstruction = new TransactionInstruction({
+    keys: [
+      {
+        pubkey: sender.publicKey,
+        isSigner: true,
+        isWritable: true
+      }
+    ],
+    programId: MEMO_PROGRAM_ID,
+    data: Buffer.from(RECOVERY_MEMO_STRING)
+  })
+  const transferInstruction =
+    await sdk.services.paymentRouterClient.createTransferInstruction({
+      total: amount,
+      sourceWallet: sender.publicKey,
+      mint: 'USDC'
+    })
+  const routeInstruction =
+    await sdk.services.paymentRouterClient.createRouteInstruction({
+      total: amount,
+      splits: [
+        {
+          amount,
+          wallet: userBank
+        }
+      ],
+      mint: 'USDC'
+    })
+  const transaction = await sdk.services.claimableTokensClient.buildTransaction(
+    { instructions: [memoInstruction, transferInstruction, routeInstruction] }
+  )
+  transaction.sign([sender])
+  const signature = await sdk.services.paymentRouterClient.sendTransaction(
+    transaction,
+    { skipPreflight: true }
+  )
+  return signature
+}
+
+/**
+ * Transfers tokens out of a user bank.
+ * Notes:
+ * - Including a signer will mark this transfer as a "withdrawal preparation"
+ *   by signing a memo indicating such. This prevents the transfer from showing
+ *   as a withdrawal on the withdrawal history page.
+ * - Users have restrictions on creating token accounts via relay, so if the
+ *   destination token account doesn't exist this might fail.
+ */
+export const transferFromUserBank = async ({
+  sdk,
+  mint,
+  connection,
+  amount,
+  ethWallet,
+  destinationWallet,
+  track,
+  make,
+  analyticsFields,
+  signer
+}: {
+  sdk: AudiusSdk
+  mint: PublicKey
+  connection: Connection
+  amount: number
+  ethWallet: string
+  destinationWallet: PublicKey
+  track: CommonStoreContext['analytics']['track']
+  make: CommonStoreContext['analytics']['make']
+  analyticsFields: any
+  signer?: Keypair
+}) => {
+  let isCreatingTokenAccount = false
+  try {
+    const instructions: TransactionInstruction[] = []
+
+    const destination = getAssociatedTokenAddressSync(mint, destinationWallet)
+
+    try {
+      await getAccount(connection, destination)
+    } catch (e) {
+      // Throws if token account doesn't exist or account isn't a token account
+      isCreatingTokenAccount = true
+      console.debug(e)
+      console.debug(
```

Comment:
Didn't know you were so anti template string :grin:

Reply:
Why use many characters when few do trick!

I should change this though I don't know if RN console works the same.

Reply:
yeah I believe RN console doesn't let you do variadic args